R/listing_export.R

```diff
@@ -0,0 +1,128 @@
+#' Export as plain text with page break symbol
+#'
+#' @inheritParams paginate_listing
+#' @param file character(1). File to write.
+#' @param paginate logical(1). Should \code{tt} be paginated before writing the file.
+#' Defaults to `TRUE` if any sort of page dimension is specified.
+#' @param \dots Passed directly to \code{\link{paginate_listing}}
+#' @param page_break character(1). Page break symbol (defaults to outputting \code{"\\s"}).
+#' @return \code{file} (this function is called for the side effect of writing the file.
+#'
+#' @note When specified, `font_size` is used *only* to determine pagination based
+#' on page dimensions. The written file is populated in raw ASCII text, which
+#' does not have the concept of font size.
+#'
+#' @export
+#'
+#'
+#' @examples
+#'
+#' dat <- ex_adae
+#' lsting <- as_listing(dat[1:25, ], key_cols = c("USUBJID", "AESOC")) %>%
+#'   add_listing_col("AETOXGR") %>%
+#'   add_listing_col("BMRKR1", format = "xx.x") %>%
+#'   add_listing_col("AESER / AREL", fun = function(df) paste(df$AESER, df$AREL, sep = " / "))
+#' main_title(lsting) <- "this is some title"
+#' main_footer(lsting) <- "this is some footer"
+#' cat(export_as_txt(lsting, file = NULL, paginate = TRUE))
+#'
+export_as_txt <- function(tt, file = NULL,
+                          page_type = NULL,
+                          landscape = FALSE,
+                          pg_width = page_dim(page_type)[if (landscape) 2 else 1],
+                          pg_height = page_dim(page_type)[if (landscape) 1 else 2],
+                          font_family = "Courier",
+                          font_size = 8, # grid parameters
+                          paginate = .need_pag(page_type, pg_width, pg_height, lpp, cpp),
+                          cpp = NULL,
+                          lpp = NULL,
+                          ..., page_break = "\\f",
```

Comment:
it is `\f` instead of `\\f` by the way

Reply:
use "\\f" will output "\f", double slash for the encoding. :) I will address this as a seperate PR per discussion we had this morning